File: week1/alfazya021/1-TwoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let dp = new Map()
    
    for(let i = 0; i < nums.length; i++) {
        let value = dp.get(target-nums[i])
        if(value !== undefined && i != value) return [value,i]
        dp.set(nums[i],i)
    }
    
    return []
};

File: week1/furkanunsalan/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_map = {}
        
        for i in range(len(nums)):
            diff = target - nums[i]
            if diff in num_map:
                return [num_map[diff], i]
            num_map[nums[i]] = i